docs/source/howto/plugins.rst

```diff
@@ -4,40 +4,312 @@
 
-.. _how-to:plugins:develop:
+.. _packages: https://docs.python.org/2/tutorial/modules.html?highlight=package#packages
 
-Developing a plugin package
-===========================
 
-`#4008`_
+Quickstart
+----------
+
+The fastest way to jumpstart an AiiDA plugin package is to use the `AiiDA plugin cutter <plugin-cutter>`_ in order to template the basic folder structure, already customized according to the desired name of your plugin, follwing AiiDA conventions.
+
+Simply go to the `AiiDA plugin cutter <plugin-cutter>`_ and follow the usage instructions.
+See also the `aiida-diff`_ demo plugin package for an in-depth explanation of the files & folders produced by the plugin cutter.
+
+In the following, we explain a few steps taken by the AiiDA plugin cutter.
+
+
+Choosing a name
+----------------
+
+The naming convention for AiiDA plugin packages is ``aiida-mycode`` for the plugin distribution on `PyPI`_ and ``aiida_mycode`` for the corresponding python package, leading to the following folder structure::
+
+   aiida-mycode/
+      aiida_mycode/
+         __init__.py
+
+.. note::
+
+   Python package names cannot contain dashes, thus the underscore.
+
+If you intend to eventually publish your plugin package, please go to the `AiiDA plugin registry <registry>`_  and choose a name that is not already taken.
+You are also encouraged to pre-register your package (instructions provided on the registry), both to reserve your plugin name and to inform others of your ongoing development.
+
+
+
+Folder structure
+----------------
+
+While it is up to you to decide the folder structure for your plugin, here is how a typical AiiDA plugin package may look like (see also the `aiida-diff`_ demo plugin)::
+
+   aiida-mycode/           - distribution folder
+      aiida_mycode/        - toplevel package (from aiida_mycode import ..)
+         __init__.py
+         calculations/
+            __init__.py
+            mycode.py      - contains MycodeCalculation
+         parsers/
+            __init__.py
+            mycode.py      - contains MycodeParser
+         data/
+            __init__.py
+            mydat.py       - contains MyData (supports code specific format)
+         commands/
+            __init__.py
+            mydat.py       - contains visualization subcommand for MyData
+         workflows/
+            __init__.py
+            mywf.py        - contains a basic workflow using mycode
+         ...
+      setup.py             - install script
+      setup.json           - install configuration
+      ...
+
+A minimal plugin package instead might look like::
+
+   aiida-minimal/
+      aiida_minimal/
+         __init__.py
+         simpledata.py
+      setup.py
+      setup.json
+
+.. _how-to:plugins:entrypoints:
+
+Adding entry points
```

Comment:
Concerning the naming discussion in #4006:

How about something like "Making the AiiDA daemon find the plugin package"?

Reply:
This is one of the problems when not registering entry points, but mostly applies to `CalcJobs` and `Workflows`. However, if you don't register a `Data` plugin, it also can cause problems.

Reply:
Enabling (Activating?) a plugin package in AiiDA?

Reply:
The main problem IMO is "adding entry points" doesn't make it clear why / if one should read this section, unless you already know what an entry point is.

Reply:
I agree with the concern, and suggested "Registering plugins through entry points" [here]
(https://github.com/aiidateam/aiida-core/issues/4006#issuecomment-630003228)

Perhaps this can be a compromise?